File: algorithms/quickSort.js
function quickSort(array) {
    if (array.length <= 1) return array

    let pivot = array[array.length - 1]
    let left = []
    let right = []
    for (let i = 0; i < array.length - 1; i++) {
        if (array[i] < pivot) {
            left.push(array[i])
        } else {
            right.push(array[i])
        }
    }
    return [...quickSort(left), pivot, ...quickSort(right)]
}

console.log(quickSort([45, 76, 23, 987, 34, 86, 2, 8742, 2, 7, 8, 32, 14, 5, 7, 435, ]))
console.log(quickSort([56, 1, 668, 45, 2, 6, 8, 9, 4, 3, 678, 34, 2, 56, 6, 433, 5]))
console.log(quickSort([56, 23, 98, 345, 76, 24, 86, 12, 6, 14, 59, 4, -24, 75]))

File: python_algorithms/quick_sort.py
number = [3,6,4,2,9,6,57,38,4,3,7824,46,2]

def quick_sort(n):
        
    if len(n) <= 1:
        return n
    else:
        pivot = n.pop()
        left = []
        right = []
        for item in n:
            if item < pivot:
                left.append(item)
            else:
                right.append(item)
        return quick_sort(left) + [pivot]+ quick_sort(right)


print(quick_sort(number))
